### DISCUSSION TITLE
Setting Wait DI for UR10e Based on TCP Force

### DISCUSSION THREAD

**Discussion by WiesenDX** [visose/Robots#235] [Q&A] [answered]
Hi guys, what I'm trying to do is let TCP descend and fit onto a plane in force mode, and stop force mode after feeling a certain amount of force from TCP.
I'm looking for the script manual and figure out how to enter and end  force mode. I also find the code to get force on TCP, but I don't know how to let it return a digital output when force on TCP reach a certain amount so I can set a wait DI.
Attaching the code for getting force on TCP.
I think maybe this is a very generic question for all robots, looking forward your help!
Thank you guys!
![20230623112812](https://github.com/visose/Robots/assets/127881362/b37318db-eba2-4bb5-98f5-2f1a8843ea7f)
![20230623112820](https://github.com/visose/Robots/assets/127881362/40c64634-5012-4322-9dfe-a84f5cbd24f2)

**Comment by visose** [comment, marked as answer]
Hi @WiesenDX, for UR specific questions that are unrelated to this plugin you might have more luck asking on a UR robots forum.
